quixstreams/core/stream/stream.py

```diff
@@ -202,53 +206,194 @@ def add_transform(
         return self._add(TransformFunction(func, expand=expand))
 
     def diff(
-        self,
-        other: "Stream",
+        self, other: "Stream", prune: bool = False, enforce_direct_split: bool = False
```

Comment:
Do we need `prune` as a parameter here?
From the usages, it seems it's always `True` in the `StreamingDataFrame.__getitem__` and `__setitem__` methods.
In this case, we may always prune the diffs until we need to do otherwise.